core-bundle/src/Cron/Cron.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\Entity\Cron as CronEntity;
+use Contao\CoreBundle\Repository\CronRepository;
+use Cron\CronExpression;
+use Psr\Log\LoggerInterface;
+
+class Cron
+{
+    /**
+     * @var string
+     */
+    public const SCOPE_WEB = 'web';
+
+    /**
+     * @var string
+     */
+    public const SCOPE_CLI = 'cli';
+
+    /**
+     * @var CronRepository
+     */
+    private $repository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var array
+     */
+    private $crons = [];
+
+    public function __construct(CronRepository $repository, LoggerInterface $logger = null)
+    {
+        $this->logger = $logger;
+        $this->repository = $repository;
+    }
+
+    /**
+     * Add a cron service.
+     *
+     * @param string $interval The interval as a CRON expression
+     */
+    public function addCronJob($service, string $method, string $interval, int $priority = 0, string $scope = null): void
```

Comment:
is `$service` a `callable` or `object` (as of PHP 7.2 if I'm right)? Maybe we pass use a `callable` instead (which in most cases will be just the service object for `__invoke`, or an array of object and method)?

Reply:
Hmm... well I need the method name for the `name` in the table anyway. Or should we ommit `::method` for the name, if the object is a `callable` (and thus only save the class name as the name of the cron job)?